bennu-scheduler/src/main/java/org/fenixedu/bennu/scheduler/log/FileSystemLogRepository.java

```diff
@@ -252,6 +253,8 @@ private static void write(String path, byte[] bytes, boolean append) {
     }
 
     private static Optional<JsonObject> readJson(String path) {
-        return read(path).map(bytes -> parser.parse(new String(bytes, StandardCharsets.UTF_8)).getAsJsonObject());
+        final Optional<JsonElement> e = read(path).map(bytes -> parser.parse(new String(bytes, StandardCharsets.UTF_8)));
+        return e.isPresent() && !e.get().isJsonNull() ? e.map(o -> o.getAsJsonObject()) : Optional.empty();
```

Comment:
can be replaced by `e.filter(o -> !o.isJsonNull()).map(JsonElement::getAsJsonObject)`

Reply:
Nice. :)